### PR TITLE
Fix installation of Qt6/WASM arch on windows + multiplatform CI tests

### LINKED ISSUE #583: Qt6 WASM installation broken
**Describe the bug**
Unable to install Qt6 WASM on windows

**To Reproduce**
`aqt install-qt windows desktop 6.2.4 wasm_32`

**Expected behavior**
Installation of Qt6.2.4 for WASM

**`aqt` output**
```
C:\Temp>aqt list-qt windows desktop --extension wasm
5.13.1 5.13.2
5.14.0 5.14.1 5.14.2
5.15.0 5.15.1 5.15.2
6.2.0 6.2.1 6.2.2 6.2.3 6.2.4
6.3.0 6.3.1 6.3.2
6.4.0

C:\Temp>aqt list-qt windows desktop --extension wasm --arch 6.2.4
wasm_32

C:\Temp>aqt install-qt windows desktop 6.2.4 wasm_32
INFO    : aqtinstall(aqt) v3.0.1 on Python 3.9.5 [CPython MSC v.1928 64 bit (AMD64)]
INFO    : Downloading qtbase...
INFO    : Downloading qtdeclarative...
INFO    : Downloading qtsvg...
INFO    : Downloading qttools...
INFO    : Redirected: mirrors.ukfast.co.uk
INFO    : Redirected: mirrors.ukfast.co.uk
INFO    : Redirected: mirrors.ukfast.co.uk
INFO    : Redirected: mirrors.ukfast.co.uk
INFO    : Finished installation of qtsvg-Windows-Windows_10_21H2-Clang-Windows-WebAssembly-X86_64.7z in 0.87389260
INFO    : Finished installation of qttools-Windows-Windows_10_21H2-Clang-Windows-WebAssembly-X86_64.7z in 0.89098170
INFO    : Downloading qttranslations...
INFO    : Redirected: mirrors.ukfast.co.uk
INFO    : Finished installation of qttranslations-Windows-Windows_10_21H2-Clang-Windows-WebAssembly-X86_64.7z in 2.53685250
INFO    : Finished installation of qtdeclarative-Windows-Windows_10_21H2-Clang-Windows-WebAssembly-X86_64.7z in 6.53476390
INFO    : Finished installation of qtbase-Windows-Windows_10_21H2-Clang-Windows-WebAssembly-X86_64.7z in 8.75650590
ERROR   : This function should not be called in this way!
Traceback (most recent call last):
  File "c:\python\python39\lib\site-packages\aqt\installer.py", line 108, in run
    args.func(args)
  File "c:\python\python39\lib\site-packages\aqt\installer.py", line 351, in run_install_qt
    Updater.update(target_config, base_path, expect_desktop_archdir)
  File "c:\python\python39\lib\site-packages\aqt\updater.py", line 286, in update
    else default_desktop_arch_dir(os_name, version)
  File "c:\python\python39\lib\site-packages\aqt\metadata.py", line 410, in default_desktop_arch_dir
    raise NotImplementedError("This function should not be called in this way!")
NotImplementedError: This function should not be called in this way!
ERROR   : aqtinstall(aqt) v3.0.1 on Python 3.9.5 [CPython MSC v.1928 64 bit (AMD64)]
Working dir: `C:\Temp`
Arguments: `['C:\\Python\\Python39\\Scripts\\aqt', 'install-qt', 'windows', 'desktop', '6.2.4', 'wasm_32']` Host: `uname_result(system='Windows', node='SPFLUBE', release='10', version='10.0.19044', machine='AMD64')`
```

**Desktop (please complete the following information):**
 - OS: Win10x64
 - `aqt` version `INFO    : aqtinstall(aqt) v3.0.1 on Python 3.9.5 [CPython MSC v.1928 64 bit (AMD64)]`

**Additional context**
Broken also on Qt6.4.0. Qt5 works, tested with 5.15.2

### PR DESCRIPTION
**Pull request by ddalcino**
This PR is intended to fix #583. It will:
* Add CI build jobs to the Azure Pipelines test suite for WASM on all three platforms, one each for Qt 5 and Qt 6.
* Eliminate the code that raises a `NotImplementedError` exception, as described in #583.
* Change the `--autodesktop` flag so that it will apply to Qt6/WASM installations. Qt6 for WASM, unlike Qt5, requires a standard desktop installation of Qt alongside the WASM installation, otherwise the installation will not be functional. This change ensures that `aqt` will warn the user if a required desktop installation is not present, and it will cause the `--autodesktop` flag to add the required architectures to the installation.
* Add an `Info.plist` file to the `accelbubble` example. The IOS builds started failing when I started this PR, probably due to a change in the MacOS-12 CI environment. I had to debug this issue to make sure that this PR is not responsible for breaking the IOS builds. This change could arguable be split off into its own PR.

### CODE REVIEW

File: aqt/metadata.py [hunk 1]
```diff
@@ -395,19 +395,17 @@ def get_arch_dir_name(host: str, arch: str, version: Version) -> str:
...
-        elif host == "mac":
-            return "macos" if version in SimpleSpec(">=6.1.2") else "clang_64"
-        else:  # Windows
-            raise NotImplementedError("This function should not be called in this way!")
+    def default_linux_desktop_arch_dir() -> str:
+        return "gcc_64"
+
+    @staticmethod
+    def default_mac_desktop_arch_dir(version: Version) -> str:
+        return "macos" if version in SimpleSpec(">=6.1.2") else "clang_64"
```
[the remark this answers is not in the document]
Comment (ddalcino) [reply aqt/metadata.py:408]: @sonatype-lift ignore

[the remark this answers is not in the document]
Comment (ddalcino) [reply aqt/metadata.py:408]: @sonatype-lift ignore

File: aqt/installer.py [hunk 2]
```diff
@@ -970,37 +970,38 @@ def _validate_version_str(
...
-            else:
-                self.logger.warning(
-                    f"You are installing the {target} version of Qt, which requires that the desktop version of Qt "
-                    f"is also installed. You can install it with the following command:\n"
-                    f"          `aqt install-qt {host} desktop {version} {default_desktop_arch}`"
-                )
-                return expected_desktop_arch_path.name, None
-        else:
-            # We do not need to worry about the desktop directory if target is not mobile.
+        is_wasm_qt6 = target == "desktop" and is_wasm and version >= Version("6.0.0")
```
[the remark this answers is not in the document]
Comment (ddalcino) [reply aqt/installer.py:976]: @sonatype-lift ignore

**Comment by ddalcino** [comment]
I think this is feature working properly now. I have `wasm_32` builds running successfully on all 3 platforms, using both Qt 5.14.2 and Qt 6.4.0.

I had to change the `--autodesktop` flag to make this work. Qt6 for WASM does not ship with its own `qmake` binary the way Qt5 does; therefore any time you install it, it's a broken installation unless you add one of the usual desktop installations. The whole point of the `--autodesktop` flag was to make this process automatic, and I think that means that it needs to work with Qt6/wasm as well. I consider this a bug fix, not a new feature.

I am not sure what's happening with the failed Mac ios builds. It's possible that this PR is responsible for them, although I do not see how. I think that something must have changed in the Macos-12 environment.

Here's the error message for the failing ios builds:

> error: Cannot code sign because the target does not have an Info.plist file and one is not being generated automatically. Apply an Info.plist file to the target using the INFOPLIST_FILE build setting or generate one automatically by setting the GENERATE_INFOPLIST_FILE build setting to YES (recommended). (in target 'accelbubble' from project 'accelbubble')

**Comment by VSRonin** [comment]
The need for the `--autodesktop` flag should probably be added to the documentation in the section that focuses on [WASM](https://aqtinstall.readthedocs.io/en/latest/getting_started.html#installing-qt-for-wasm)

File: aqt/updater.py [hunk 3]
```diff
@@ -279,12 +278,14 @@ def update(cls, target: TargetConfig, base_path: Path, installed_desktop_arch_di
...
-                    else default_desktop_arch_dir(os_name, version)
-                )
+            else:  # qt6 mobile or wasm
+                if installed_desktop_arch_dir is not None:
+                    desktop_arch_dir = installed_desktop_arch_dir
+                else:
+                    # Use MetadataFactory to check what the default architecture should be
+                    meta = MetadataFactory(ArchiveId("qt", os_name, "desktop"))
+                    desktop_arch_dir = meta.fetch_default_desktop_arch(version)
+                    # NOTE: desktop_arch_dir may need to be wasm_32 in the wasm case
```
Comment (ddalcino) [inline aqt/updater.py:288]:
```suggestion
```

File: aqt/updater.py [hunk 4]
```diff
@@ -279,12 +278,14 @@ def update(cls, target: TargetConfig, base_path: Path, installed_desktop_arch_di
...
-                    if installed_desktop_arch_dir is not None
-                    else default_desktop_arch_dir(os_name, version)
-                )
+            else:  # qt6 mobile or wasm
+                if installed_desktop_arch_dir is not None:
+                    desktop_arch_dir = installed_desktop_arch_dir
+                else:
+                    # Use MetadataFactory to check what the default architecture should be
+                    meta = MetadataFactory(ArchiveId("qt", os_name, "desktop"))
+                    desktop_arch_dir = meta.fetch_default_desktop_arch(version)
```
Comment (ddalcino) [inline aqt/updater.py:287]: New uncovered lines of code: these deserve a unit test

**Comment by miurahr** [comment]
It is great that we have the fix, test cases, CI test config, and document with the example... Perfect.
I cannot find any point I need to ask.

**Comment by VSRonin** [comment]
Can I ask what is holding the merge?

**Pull request merged**